virtManager/object/domain.py

```diff
@@ -436,6 +436,11 @@ def has_nvram(self):
         return bool(self.get_xmlobj().is_uefi() or
                     self.get_xmlobj().os.nvram)
 
+    def has_tpm_state(self):
+        for tpm in self.get_xmlobj().devices.tpm:
+            if tpm.type == "emulator":
+                return True
```

Comment:
please explicitly return `False` at the end of the function, otherwise it returns `None`

another possibility is to use `any()` -- something like:
```python
return any(tpm.type == "emulator" tpm in self.get_xmlobj().devices.tpm)
```

Reply:
I'll use the oneliner, thanks. My primary complaint with oneliners is they obscure code coverage testing but it doesn't matter in this instance